enterprise/cmd/frontend/internal/authz/gitlab/oauth.go

```diff
@@ -0,0 +1,42 @@
+package gitlab
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/gitlab"
+)
+
+// FetchUserPerms returns a list of project IDs (on code host) that the given account
```

Comment:
```suggestion
// FetchUserPerms returns a list of private project IDs (on code host) that the given account
```

I would move this here because it's an important if I understood it correctly: the list doesn't contain public projects, yes?